py/src/braintrust/logger.py

```diff
@@ -2394,13 +2395,34 @@ def __enter__(self):
             self._context_token = _state.current_span.set(self)
         return self
 
-    def __exit__(self, type, value, callback):
-        del type, value, callback
+    def __exit__(self, exc_type, exc_value, tb):
+        try:
+            if exc_type is not None:
+                self.log_internal(dict(error=stringify_exception(exc_type, exc_value, tb)))
+        finally:
+            if self.set_current:
+                _state.current_span.reset(self._context_token)
 
-        if self.set_current:
-            _state.current_span.reset(self._context_token)
+            self.end()
+
+
+def stringify_exception(exc_type, exc_value, tb):
+    lines = traceback.format_exception(exc_type, exc_value, tb)
```

Comment:
Python exposes more granular methods to format individual parts of the stack trace. You could write something like this:
```
return ''.join(traceback.format_exception_only(exc_type, exc_value) + ['\nTraceback (most recent call last):\n'] + traceback.format_tb(tb))
```

Reply:
nice thanks